designer/server/src/test/scala/pl/touk/nussknacker/ui/api/ScenarioParametersApiSpec.scala

```diff
@@ -0,0 +1,93 @@
+package pl.touk.nussknacker.ui.api
+
+import io.restassured.RestAssured.`given`
+import io.restassured.module.scala.RestAssuredSupport.AddThenToResponse
+import org.scalatest.freespec.AnyFreeSpecLike
+import pl.touk.nussknacker.test.{
+  NuRestAssureExtensions,
+  NuRestAssureMatchers,
+  PatientScalaFutures,
+  RestAssuredVerboseLogging
+}
+import pl.touk.nussknacker.ui.api.helpers.{NuItTest, NuTestScenarioManager}
+
+class ScenarioParametersApiSpec
```

Comment:
1. What do you mean by missing tests for authorization? There are tests for: access to all categories, access to one category, access to none category.
2. Regarding engine errors:
- It will be hard to do this at this level because it would require showing different combinations of application setup and currently we don't have option to change this in a given part
- But it will be easy to do at the domain service layer and I have a plan to add them in the next PRs. Currently, engine errors are always an empty list because of FIXME in `ProcessingTypeData:124`

Reply:
1. Ah, I mean authentication. You can write test without basic authentication.

Reply:
Ah, you are right. I added it.